partner_manual_rank/models/res_partner.py

```diff
@@ -72,3 +72,12 @@ def _default_is_customer(self):
 
     def _default_is_supplier(self):
         return self.env.context.get("res_partner_search_mode") == "supplier"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if "customer_rank" in vals and "is_customer" not in vals:
```

Comment:
Shouldn't this be covered by the compute method?

Reply:
> Shouldn't this be covered by the compute method?

No because you defined a default value for the `is_customer` and `is_supplier` field. Therefore, if these fields are not part of the dict, the orm call the write method with the default values . As a side effect, the inverse methods defined for these fields are called and overwrite the value provided for the 'x_rank' fields if they are equal to '1'. If you remove the override if the create method and launch the tests you'll see the problem.